internal/repository/metrics.go

```diff
@@ -0,0 +1,237 @@
+	return metric, nil
+}
+
+func (r *MetricRepositoryImpl) Read(
+	ctx context.Context,
+	metricID string,
+) (*model.Metric, error) {
+	qu, _, err := goqu.
+		Select(&model.Metric{}).
+		From(metricTName).
+		Where(goqu.Ex{"id": metricID}).
+		Limit(1).
+		ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("read metric error during query building: %w", err)
+	}
+
+	var newMetric model.Metric
+	row := r.gr.DB.QueryRowxContext(ctx, qu)
+	err = row.StructScan(&newMetric)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, fmt.Errorf("read metric error during scan row: %w", err)
+	}
+
+	if err == sql.ErrNoRows {
+		return nil, nil
+	}
+
+	return &newMetric, nil
+}
+
+func (r *MetricRepositoryImpl) ReadIDs(ctx context.Context) (*[]string, error) {
+	qu, _, err := goqu.
+		Select("id").
+		From(metricTName).
+		ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("read ids metric error during query building: %w", err)
+	}
+
+	rows, err := r.gr.DB.QueryxContext(ctx, qu)
+	if err != nil {
+		return nil, fmt.Errorf("read metrics error during querying: %w", err)
+	}
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("read metrics error during querying: %w", err)
+	}
+	defer rows.Close()
+
+	var ids []string
+	for rows.Next() {
+		id := ""
+		err := rows.Scan(&id)
+		if err != nil {
+			return nil, fmt.Errorf("read metrics error during scan rows: %w", err)
+		}
+		ids = append(ids, id)
+	}
+
+	if err == sql.ErrNoRows {
+		return nil, nil
+	}
+
+	return &ids, nil
+}
+
+func (r *MetricRepositoryImpl) ReadMany(
+	ctx context.Context,
+	metricIDs []string,
+) (*[]model.Metric, error) {
+	inIDs := []any{}
+	for _, id := range metricIDs {
+		inIDs = append(inIDs, id)
+	}
+
+	qu, _, err := goqu.
+		Select(&model.Metric{}).
+		From(metricTName).
+		Where(goqu.C("id").In(inIDs...)).
+		ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("read metrics error during query building: %w", err)
+	}
+
+	rows, err := r.gr.DB.QueryxContext(ctx, qu)
+	if err != nil {
+		return nil, fmt.Errorf("read metrics error during querying: %w", err)
+	}
+	if rows.Err() != nil {
+		return nil, fmt.Errorf("read metrics error during querying: %w", err)
+	}
+	defer rows.Close()
+
+	newMetrics := []model.Metric{}
+	for rows.Next() {
+		metric := model.Metric{}
+		err := rows.StructScan(&metric)
+		if err != nil {
+			return nil, fmt.Errorf("read metrics error during scan rows: %w", err)
+		}
+		newMetrics = append(newMetrics, metric)
+	}
+
+	if err == sql.ErrNoRows {
+		return nil, nil
+	}
+
+	return &newMetrics, nil
+}
+
+func (r *MetricRepositoryImpl) Update(
+	ctx context.Context,
+	metric *model.Metric,
+) (*model.Metric, error) {
+	qu, _, err := goqu.
+		Update(metricTName).
+		Set(metric).
+		Where(goqu.Ex{"id": metric.ID}).
+		ToSQL()
+	if err != nil {
+		return nil, fmt.Errorf("create metric error during query building: %w", err)
+	}
+
+	tx, err := r.gr.BeginTxx(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer tx.Commit()
```

Comment:
```suggestion
	defer tx.Rollback()
```
`tx.Commit()` дергается только в случае отсутствия ошибки в конце работы с вызовами БД

Но в данном случае вообще нет смысла открывать транзакцию, так как в ее рамках будет выполняться всего один запрос, он и так выполнится транзакционно-безопасно.